qoffer-infrastructure/src/main/groovy/life/qbic/portal/qoffer2/customers/CustomerDatabaseQueries.groovy

```diff
@@ -0,0 +1,131 @@
+package life.qbic.portal.qoffer2.customers
+
+import life.qbic.datamodel.people.Person
+import life.qbic.datamodel.persons.Affiliation
+import life.qbic.portal.portlet.customers.Customer
+import life.qbic.portal.qoffer2.database.DatabaseSession
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+
+import java.sql.Connection
+import java.sql.PreparedStatement
+import java.sql.ResultSet
+import java.sql.SQLException
+
+/**
+ * This class contains all queries on the customer database
+ *
+ * All database queries for the customer database are collected here. Only here the {@link DatabaseSession} is used to connect
+ * to the database
+ *
+ * @since: 1.0.0
+ * @author: Jennifer Bödker
+ *
+ */
+class CustomerDatabaseQueries {
+
+    private final Connection databaseConnection
+
+    private static final Logger LOG = LogManager.getLogger(CustomerDatabaseQueries.class)
+
+    CustomerDatabaseQueries(Connection connection){
+        this.databaseConnection = connection
+    }
+
+    /**
+     * Searches for a customer based on its last name
+     *
+     * @param lastName of the customer
+     * @return a list of customers with a matching last name
+     */
+    List<Person> findPersonByName(String lastName){
+        List<Person> res = []
+        try{
+            String sql = "SELECT id, first_name, family_name, email from persons WHERE family_name = ?"
+
+            PreparedStatement statement = null
+            try{
+                statement = databaseConnection.prepareStatement(sql)
+                statement.setString(1, lastName)
+                ResultSet rs = statement.executeQuery()
+                println rs
+
+                while (rs.next()) {
+                    fetchAffiliationForPerson(rs.getString(1).toInteger())
+                    res <<  new Person(rs.getString(2),rs.getString(3),rs.getString(4))
+                }
+
+                return res
+            } catch (SQLException e) {
+                LOG.error("SQL operation unsuccessful: " + e.getMessage())
+                e.printStackTrace()
+            } finally {
+                DatabaseSession.logout(databaseConnection)
+            }
+            return null
```

Comment:
If the database operation was unsuccessful, the Connector i suggest that the method throws an Exception defined in the domain layer, like `DatabaseQueryException`, or similar

Reply:
107fa8a